testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/organization/authentication/OrganizationAuthenticationTest.java

```diff
@@ -134,4 +140,43 @@ public void testForceReAuthenticationBeforeRequiredAction() {
             oauth.maxAge(null);
         }
     }
+
+    @Test
+    public void testRequiresUserMembership() {
+        runOnServer(setAuthenticatorConfig(OrganizationAuthenticatorFactory.REQUIRES_USER_MEMBERSHIP, Boolean.TRUE.toString()));
+
+        try {
+            OrganizationResource organization = testRealm().organizations().get(createOrganization().getId());
+            UserRepresentation member = addMember(organization);
+            organization.members().member(member.getId()).delete().close();
+            oauth.clientId("broker-app");
+            loginPage.open(bc.consumerRealmName());
+            loginPage.loginUsername(member.getEmail());
+            // user is not a member of any organization
+            assertThat(errorPage.getError(), Matchers.containsString("User is not a member of the organization " + organization.toRepresentation().getName()));
+
+            organization.members().addMember(member.getId()).close();
+            OrganizationRepresentation orgB = createOrganization("org-b");
+            oauth.clientId("broker-app");
+            oauth.scope("organization:org-b");
+            loginPage.open(bc.consumerRealmName());
+            loginPage.loginUsername(member.getEmail());
+            // user is not a member of the organization selected by the client
+            assertThat(errorPage.getError(), Matchers.containsString("User is not a member of the organization " + orgB.getName()));
+            errorPage.assertTryAnotherWayLinkAvailability(false);
+        } finally {
```

Comment:
I would also add successful login testcase:
```
oauth.clientId("broker-app");
oauth.scope("organization:neworg");
loginPage.open(bc.consumerRealmName());
loginPage.loginUsername(member.getEmail());
loginPage.login(memberPassword);
appPage.assertCurrent();
```

Reply:
We already have tests covering successful use case, don't we?